### DISCUSSION TITLE
Service Dependency Graph for Jaeger

### DISCUSSION THREAD

**Discussion by nicolastakashi** [grafana/grafana#52035] [Dashboards, Panels & Visualizations]
As discussed on this [issue](https://github.com/grafana/grafana/issues/50254) would be really nice to have Service Graph available for Jaeger as well.

Since the Node Graph is not coupled to DataSource,  just needs some work converting whatever metrics/data Jaeger exposes for their service dependency graph into the format the node graph panel needs.

**Comment by connorlindsey** [comment]
This is a great idea. @nicolastakashi If you'd be interested in implementing this, that'd be amazing as I'm not sure when we can fit it into our current work. A rough outline of the implementation would be:
1. Add a 'Service Graph' query type to the Jaeger explore page
2. When run, query Jaeger for their service dependency graph metrics
3. Format those into dataframes according to the [Node Graph api](https://grafana.com/docs/grafana/latest/visualizations/node-graph/#data-requirements) - 1 frame for nodes, 1 for edges
4. Optionally, add any useful links to the dataframe config

Link to the internal Jaeger api (https://www.jaegertracing.io/docs/1.35/apis/#service-dependencies-graph-internal)

**Reply by mszabo-wikia** [reply]
I took a stab at this: https://github.com/grafana/grafana/pull/72200

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be moving this discussion, and you can follow it [here](https://github.com/grafana/grafana/issues/79372). Sorry for the inconvenience, and thanks for your contributions!
